pypeit/spectrographs/gemini_gmos.py

```diff
@@ -482,10 +489,213 @@ def allowed_mosaics(self):
+            slitid=np.array(slitID, dtype=int),
+            align=mask_tbl['priority'].value == b'0',
+            science=mask_tbl['priority'].value != b'0',
+            onsky=np.array([
+                slit_ra, slit_dec, 
+                np.array(mask_tbl['slitsize_y'].to('arcsec').value, dtype=float),
+                np.array(mask_tbl['slitsize_x'].to('arcsec').value, dtype=float),
+                slit_pas]).T,
+           objects=objects,
+           mask_radec=(mask_coord.ra.deg, mask_coord.dec.deg),
+           posx_pa=posx_pa)
+        return self.slitmask
+
+    def get_maskdef_slitedges(self, ccdnum=None, filename=None, debug=None, wcs_file=None,
+                              binning=None):
+        if filename is not None:
+            self.get_slitmask(filename)
+        else:
+            msgs.error('The name of a ODF file should be provided in the PypeIt file')
+
+        # Binning of flat
+        _, bin_spat= parse.parse_binning(binning) 
+
+        # Slit center
+        slit_coords = SkyCoord(ra=self.slitmask.onsky[:,0], 
+                               dec=self.slitmask.onsky[:,1], unit='deg')
+        mask_coord = SkyCoord(ra=self.slitmask.mask_radec[0],
+                              dec=self.slitmask.mask_radec[1], unit='deg')
+
+        # Load up the acquisition image (usually a sciframe)
+        hdul_acq = fits.open(wcs_file)
+        acq_binning = self.get_meta_value(self.get_headarr(hdul_acq), 'binning')
+        _, bin_spat_acq = parse.parse_binning(acq_binning) 
+        wcss = [wcs.WCS(hdul_acq[i].header) for i in range(1, len(hdul_acq))]
+
+        left_edges = []
+        right_edges = []
+        for islit in range(self.slitmask.nslits):
+            # DEBUGGING
+            #islit = 14  # 10043
+            # Left coord
+            left_coord = slit_coords[islit].directional_offset_by(
+                self.slitmask.onsky[islit,4]*units.deg - 180.*units.deg,
+                self.slitmask.onsky[islit,2]*units.arcsec/2.)
+            right_coord = slit_coords[islit].directional_offset_by(
+                self.slitmask.onsky[islit,4]*units.deg,
+                self.slitmask.onsky[islit,2]*units.arcsec/2.)
+                
+            got_it = False
+            for kk, iwcs in enumerate(wcss):
+                pix_xy = iwcs.world_to_pixel(left_coord)
+                # Do we have the right WCS?
+                if float(pix_xy[0]) > 0 and float(pix_xy[0]) < (
+                    hdul_acq[kk+1].header['NAXIS1']-1) and not got_it:
+                    left_edges.append(float(pix_xy[1])*bin_spat_acq/bin_spat)
+                    # Right
+                    pix_xy2 = iwcs.world_to_pixel(right_coord)
+                    right_edges.append(float(pix_xy2[1])*bin_spat_acq/bin_spat)
+                    # Occasionally a slit thinks it is on 2 detectors -- this avoids that
+                    got_it = True
+                    print(f'matched to {kk}, {pix_xy}, {pix_xy2}')
```

Comment:
Instead of using `got_it`, I would put in a break, and you can combine the inequalities:
```
for kk, iwcs in enumerate(wcss):
    pix_xy = iwcs.world_to_pixel(left_coord)
    if 0 < float(pix_xy[0]) < hdul_acq[kk+1].header['NAXIS1']-1:
        ...
        print(f'matched to {kk}, {pix_xy}, {pix_xy2}')
        break
```

Reply:
good one.  always scared as to whether a `break` jumps out of the `for` loop
 if it is embedded in an `if`